ark/analysis/spatial_analysis.py

```diff
@@ -342,12 +349,40 @@ def calculate_cluster_spatial_enrichment(all_data, dist_matrices_dict, included_
         dist_mat = dist_matrices_dict[fov]
 
         # Get close_num and close_num_rand
-        close_num, pheno_nums, pheno_nums_per_id = spatial_analysis_utils.compute_close_cell_num(
+        close_num, pheno_nums = spatial_analysis_utils.compute_close_cell_num(
             dist_mat=dist_mat, dist_lim=dist_lim, analysis_type="cluster",
             current_fov_data=current_fov_pheno_data, cluster_ids=cluster_ids)
 
-        close_num_rand = spatial_analysis_utils.compute_close_cell_num_random(
-            pheno_nums, dist_mat, dist_lim, bootstrap_num)
+        # subset distance matrix with context
+        if context_col is not None:
+            close_num_rand = np.zeros((*close_num, bootstrap_num), dtype=np.uint16)
+
+            context_nums_per_id = \
```

Comment:
What is this one-liner doing?

Reply:
This creates a dictionary with context names as keys and lists of member cell's label values as values.  i.e:
```
context_nums_per_id = {
    'immune': [13, 490, 793, ...],
    'epithelial': [17, 289, 391, ...],
    ...
}
```